src/test/unit/test.js

```diff
@@ -491,6 +491,51 @@ describe('Client', function() {
     })
   })
```

Comment:
can you put the tests in functional tests and remove unit tests? we had decided move the test cases to functional tests from unit tests.